openquake/hazardlib/tests/lt_test.py

```diff
@@ -0,0 +1,128 @@
+   </hypoDepthDist>
+</pointSource>''')
+
+
+def scaling_rates(srcs):
+    return [getattr(src, 'scaling_rate', 1.) for src in srcs]
+
+
+class CollapseTestCase(unittest.TestCase):
+    def setUp(self):
+        # setup a simple logic tree with 3 realizations
+        #    ___/ b11 (w=.2)
+        #  _/   \ b12 (w=.2)
+        #   \____ b02 (w=.6)
+        self.bs0 = bs0 = lt.BranchSet('abGRAbsolute')
+        bs0.branches = [lt.Branch('bs0', 'b01', .4, (4.6, 1.1)),
+                        lt.Branch('bs0', 'b02', .6, (4.4, 0.9))]
+
+        self.bs1 = bs1 = lt.BranchSet('maxMagGRAbsolute')
+        bs1.branches = [lt.Branch('bs1', 'b11', .5, 7.0),
+                        lt.Branch('bs1', 'b12', .5, 7.6)]
+        bs0.branches[0].bset = bs1
+
+        # setup sitecol, srcfilter, gsims, imtls
+        sitecol = site.SiteCollection(
+            [site.Site(Point(0, 0), numpy.array([760.]))])
+        self.srcfilter = calc.filters.SourceFilter(
+            sitecol, {'default': 200})
+        self.gsims = [valid.gsim('ToroEtAl2002')]
+        self.imtls = DictArray({'PGA': valid.logscale(.01, 1, 5)})
+        self.sg = sourceconverter.SourceGroup(ps.tectonic_region_type, [ps])
+
+    def full_enum(self):
+        # compute the mean curve with full enumeration
+        srcs = []
+        weights = []
+        grp_id = 0
+        for weight, branches in self.bs0.enumerate_paths():
+            path = tuple(br.branch_id for br in branches)
+            bset_values = self.bs0.get_bset_values(path)
+            # first path: [(<b01 b02>, (4.6, 1.1)), (<b11 b12>, 7.0)]
+            sg = lt.apply_uncertainties(bset_values, self.sg)
+            for src in sg:
+                src.grp_id = grp_id
+            grp_id += 1
+            srcs.extend(sg)
+            weights.append(weight)
+        for i, src in enumerate(srcs):
+            src.id = i
+        pmap = classical(srcs, self.srcfilter, self.gsims,
+                         dict(imtls=self.imtls, truncation_level2=2))['pmap']
+        curves = [pmap[grp_id].array[0, :, 0] for grp_id in sorted(pmap)]
+        mean = numpy.average(curves, axis=0, weights=weights)
+        return mean, srcs, weights
+
+    def test_point_source_full_enum(self):
+        # compute the mean curve
+        mean, srcs, weights = self.full_enum()
+        assert weights == [.2, .2, .6]
+        assert scaling_rates(srcs) == [1, 1, 1]
+
+        # compute the partially collapsed curve
+        self.bs1.collapsed = True
+        coll1, srcs, weights = self.full_enum()
+        assert weights == [.4, .6]  # two rlzs
+        # self.plot(mean, coll1)
+        assert scaling_rates(srcs) == [1.0, 0.5, 0.5, 1.0]
+        numpy.testing.assert_allclose(mean, coll1, atol=.1)
+
+        # compute the fully collapsed curve
+        self.bs0.collapsed = True
+        self.bs1.collapsed = True
+        coll2, srcs, weights = self.full_enum()
+        assert weights == [1]  # one rlz
+        # self.plot(mean, coll2)
```

Comment:
leftover

Reply:
No, the comment is there on purpose, it is useful uncomment that line and look at the curves